src/utils/activity/ExecuteActivityOutputStepBase.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtTreeItem, AzureWizardExecuteStep, ExecuteActivityContext, IActionContext } from "@microsoft/vscode-azext-utils";
+import { Progress } from "vscode";
+import { tryCatchActivityWrapper } from "../../utils/activity/activityUtils";
+
+export interface ExecuteActivityOutput {
+    /**
+     * The activity child item to display on success or fail
+     */
+    item?: AzExtTreeItem;
+    /**
+     * The output log message(s) to display on success or fail
+     */
+    output?: string | string[];
+}
+
+export interface ExecuteActivityOutputOptions {
+    shouldSwallowError?: boolean;
+}
+
+/**
+ * An execute activity base step (wrapper) that automatically handles displaying activity children and/or output log messages on success or fail
+ */
+export abstract class ExecuteActivityOutputStepBase<T extends IActionContext & ExecuteActivityContext> extends AzureWizardExecuteStep<T> {
+    abstract priority: number;
+    protected options: ExecuteActivityOutputOptions = {};
+
+    protected success: ExecuteActivityOutput;
+    protected fail: ExecuteActivityOutput;
+
+    public async execute(context: T, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        this.success = this.initSuccessOutput(context);
+        this.fail = this.initFailOutput(context);
+
+        await tryCatchActivityWrapper(() => this.executeCore(context, progress), context, this.success, this.fail, this.options);
```

Comment:
Is there really any need to make `tryCatchActivityWrapper` a util function? I feel like it makes it a little bit more complicated than just having a try/catch clause in execute here. I don't think that you could really use this util function with anything other than this `ExecuteActivityOutputStepBase`.

Reply:
That's a good point.  I originally implemented it without the execute step base but forgot to extract this logic to sit together with it.